backend/layers/business/business.py

```diff
@@ -609,14 +613,30 @@ def create_collection_version(self, collection_id: CollectionId) -> CollectionVe
         added_version_id = self.database_provider.add_collection_version(collection_id)
         return self.get_collection_version(added_version_id)
 
-    def delete_collection_version(self, version_id: CollectionVersionId) -> None:
+    def delete_collection_version(self, collection_version: CollectionVersionWithDatasets) -> None:
         """
         Deletes a collection version. This method will raise an error if the version is published.
         (Note: for performance reasons, the check is performed by the underlying layer)
         """
-        self.database_provider.delete_collection_version(version_id)
-
-    def delete_dataset_versions_from_bucket(self, dataset_version_ids: List[str], bucket: str) -> List[str]:
+        unpublished_versions_of_published_datasets, unpublished_datasets, versions_to_delete_from_s3 = [], [], []
+        for dv in collection_version.datasets:
+            unpublished_versions = self.get_unpublished_dataset_versions(dv.canonical_dataset.dataset_id)
+            versions_to_delete_from_s3.extend(unpublished_versions)  # All unpublished s3 assets should be deleted
+            if dv.canonical_dataset.published_at:
+                unpublished_versions_of_published_datasets.extend(unpublished_versions)
+            else:
+                unpublished_datasets.append(dv.canonical_dataset)
+        self.delete_dataset_version_assets(versions_to_delete_from_s3)
```

Comment:
Add some more spaces in this function for readability. It'll make it easier to know if a comment refers to the line above or bellow.